src/common/bitmap.h

```diff
@@ -47,7 +47,7 @@ struct BitMap {
     for (bst_omp_uint i = 0; i < nsize; ++i) {
       uint32_t res = 0;
       for (int k = 0; k < 32; ++k) {
-        int bit = vec[(i << 5) | k];
+        uint32_t bit = vec[(i << 5) | k];
```

Comment:
cppcheck was complaining about `shifting signed 32-bit value by 31 bits is undefined behaviour`.

Reply:
Good to know. Thanks.